micromagneticmodel/driver.py

```diff
@@ -50,6 +50,10 @@ def _schedule_commands(self, system, runner):
     def _read_data(self, system):
         """Update system with simulation output (magnetisation and scalar data)."""
 
+    @abc.abstractmethod
+    def check_system(self, system):
+        """Check if the dynamics of the system is defined."""
```

Comment:
This docstring should be more generic. Checks in this function are not limited to the dynamics equation. We could e.g. also check for the energy equation or anything else that might be required.

Reply:
```suggestion
        """Check if the system contains all required information."""
```